### DISCUSSION TITLE
what does R and T in base_cma.json mean

### DISCUSSION THREAD

**Discussion by buptseelihang** [NVlabs/instant-ngp#408] [Q&A] [answered]
if R means rotaion matrix ,why it have four parameter?  what does slice and time mean

the default value of T is [0.5,0.5,2.0]? i try to change it in nerf_loder.cu but it doesn't work

how to use the transform_matrix of frames to get cameras' position and view direction so i can  render a high quality video?

THANK YOU~

**Comment by Tom94** [comment, marked as answer]
Hi there, if you have a stored camera path, you can use
```py
testbed.load_camera_path(args.camera_path)
testbed.render(start_t=time)
```
where `time` is a value from 0 to 1 that indicates the position along the path.

As for R, T, slice and time:
- R is a quaternion describing the orientation (R for rotation) of the camera,
- T is the position (T for translation) of the camera,
- slice is used to animate the slice plane when rendering in slice mode, or, alternatively, to animate the focal plane when rendering with depth of field, and
- time is the current time along the camera path at the time of saving it.

**Reply by buptseelihang** [reply]
Thank you for you replay ,i have got R use quaternion to describe rotation. I convert quaternion into rotation matrix but i find it's different from with transform_matrix in transforms.json.Is there a coordinate system conversion? it seems exchange R's row and change the sign in 2 and 3 col. But i can't find  T's contact between camera path and transforms.json .

In a word ,how to convert transfrom matrix to camera path.

**Reply by kajc10** [reply]
@buptseelihang did you manage to solve this? I'm trying to convert transforms.json coordinates to base_cam.json to render explicitly the same frames.

**Reply by 0xiNach** [reply]
@buptseelihang Answered in #490
